### ISSUE TITLE
[CHANGE] - Notify MdiItems about reopening attempts

### ISSUE THREAD

**Issue by mariusalexander** [dlr-gtlab/gtlab-core#1315]
### Summary

By default a MdiItem is created to view/edit a single data object (`GtObject`). If the same data object is opened once again a new MdiItem is created and displayed - unless the flag `MdiItem::allowsMultipleInstances` is disabled or a custom `ID` is generated that is the same as the `ID` of another already opened MdiItem. That way one can map multiple objects onto the same MdiItem. 

However, if a MdiItem is already opened and the same (or a different) data object is attempted to be opened, the existing MdiItem is **not** notified in anyway about this attempt. This should be fixed/implemented to allow a MdiItem to update its contents if desired.

This snippet would have to be modified/is the issue for the current behavior as far as I can tell
https://github.com/dlr-gtlab/gtlab-core/blob/1866efbbca5261820c9d9d25581d5e107e95788b/src/gui/gt_mdilauncher.cpp#L530-L549

### Impact on the codebase

IMO this must be implemented in such a way that it does not create a big API change or worse that the API stays the same but the behavior is changed in such a way that existing MdiItem implementations of other module do not work properly anymore (e.g. because they assumed that the `MdiItem::setData` is called only once during their lifetime).

This is probably a feature/change for version 2.1.x but it would be nice to have it in 2.0.x already because the [issue 97 in the intelligraph module](https://github.com/dlr-gtlab/intelligraph-module/issues/97) could not be implemented fully due to the current behavior

### Current workarounds

A workaround is either very hard/complex to implement or not possible in a sensible manner.
